modules/exploits/linux/local/cve_2021_3493_overlayfs.rb

```diff
@@ -0,0 +1,139 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = GreatRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Post::Linux::Priv
+  include Msf::Post::Linux::System
+  include Msf::Post::Linux::Compile
+  include Msf::Post::Linux::Kernel
+  include Msf::Post::File
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => '2021 Ubuntu Overlayfs LPE',
+        'Description' => %q{
+          2021 Ubuntu Overlayfs LPE
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'ssd-disclosure',
+          'bwatters-r7' # Aka @tychos_moose, Metasploit Module
+        ],
+        'DisclosureDate' => '2021-04-12',
+        'Platform' => [ 'linux' ],
+        'Arch' => [ ARCH_X86, ARCH_X64 ],
+        'SessionTypes' => [ 'shell', 'meterpreter' ],
+        'Targets' => [[ 'Auto', {} ]],
+        'Privileged' => true,
+        'References' => [
+          [ 'CVE', '2021-3493' ],
+          [ 'URL', 'https://ssd-disclosure.com/ssd-advisory-overlayfs-pe/' ],
+          [ 'URL', 'https://github.com/briskets/CVE-2021-3493' ]
+        ],
+        'Notes' => {
+          'Reliability' => [ REPEATABLE_SESSION ],
+          'Stability' => [ ],
+          'SideEffects' => [ ]
```

Comment:
ARTIFACTS_ON_DISK should probably be added since you're either copying up the ELF executable or you're compiling it yourself.

Reply:
Added